internal/keysign/keysign.go

```diff
@@ -0,0 +1,129 @@
+package keysign
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	tsslib "github.com/vultisig/mobile-tss-lib/tss"
+
+	"github.com/vultisig/vultisigner/config"
+	"github.com/vultisig/vultisigner/internal/logging"
+	"github.com/vultisig/vultisigner/internal/tss"
+	"github.com/vultisig/vultisigner/internal/types"
+	"github.com/vultisig/vultisigner/relay"
+)
+
+func JoinKeySign(ks *types.KeysignRequest) ([]string, error) {
+	result := []string{}
+	keyFolder := config.AppConfig.Server.VaultsFilePath
+	serverURL := config.AppConfig.Relay.Server
+
+	server := relay.NewServer(serverURL)
+
+	// Let's register session here
+	if err := server.RegisterSession(ks.Session, ks.Key); err != nil {
+		return result, fmt.Errorf("failed to register session: %w", err)
+	}
+	// wait longer for keysign start
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
```

Comment:
For keysign , TSS timeout in 1 minutes , with 3 retries , it is maximum 3 minutes , can we please set it to timeout at a little over 3 minutes, like 3minutes + 3 seconds